keps/sig-node/2535-ensure-secret-pulled-images/README.md

```diff
@@ -129,27 +143,49 @@ use un-encrypted...
+
 ## Proposal
 
-For alpha the kubelet will keep a list, across reboots of host and restart of
-kubelet, of container images that required authentication and a list of the
-authentications that successfully pulled the image.
-For beta an API will be considered to manage the ensure metadata.
+The kubelet will track container images and the list of authentication information
+that lead to their successful pulls. This data will be persisted across reboots
+of the host and restarts of the kubelet.
 
-The kubelet will ensure any image in the list is always pulled if an authentication
-used is not present, thus enforcing authentication / re-authentication.
+The persistent authentication data storage will be done using files kept in the
+kubelet directory on the node. The content of these files will be structured and
+versioned using standard config file API versioning techniques.
 
-There will be two different kubelet configuration options added, as well as a feature
-gate to gate their use:
-- `pullImageSecretRecheck`
-    - A boolean that toggles this behavior. If `false`, the kubelet will fallback to the
-      old behavior: only pull an image if it's not present.
-- `pullImageSecretRecheckPeriod`
-    - the period after which the kubelet's cache will be invalidated,
-      thus causing rechecks for all `IfNotPresent` images that are recreated.
-    - If set to `0s`, or `0`, but `pullImageSecretRecheck` is `true`, then
-      the kubelet will never invalidate its cache, but will maintain one.
+The kubelet will ensure any image requiring credential verification is always pulled if authentication information
+from an image pull is not yet present, thus enforcing authentication / re-authentication.
 
+Two new kubelet configuration options will be added, as well as a feature
+gate to gate their use:
+- `imagePullCredentialsVerificationPolicy`
+    - An enum of 4 possible values:
+      - `NeverVerify`
+        - **original behavior before this feature**
+        - anyone on a node can use any image present on the node
+      - `NeverVerifyPreloadedImages`
+        - **default value**
+        - images that were pulled to the node by something else than the kubelet can be used without reverifying pull credentials
+      - `NeverVerifyAllowlistedImages`
+        - like `NeverVerifyPreloadedImages` but only node images from `preloadedImagesVerificationAllowlist` don't require reverification
+      - `AlwaysVerify`
+        - all images pulled with credentials require credential reverification
+- `preloadedImagesVerificationAllowlist`
+  - a string list to be used with the `NeverVerifyAllowlistedImages` image pull credentials verification policy
+  - images specified by this list will not require credential reverification if they are requested
+    by a pod with `IfNotPresent` or `Never` image pull policies
+  - the list accepts wildcard suffix in a string where `*` is the wildcard character
```

Comment:
do we intend to limit the wildcard to a full path segment, or are we intending to match `example.com/foo-*`?

if the former, should be something like:
```suggestion
  - the list accepts a `/*` wildcard suffix
```